packages/client/src/client.ts

```diff
@@ -92,12 +93,57 @@ export class TypedAmqpClient<TContract extends ContractDefinition> {
     };
 
     const publishMessage = (validatedMessage: unknown): Future<Result<void, TechnicalError>> => {
+      const compressionAlgorithm = publisher.compression;
+      const publishOptions: Options.Publish = { ...options };
+
+      // Apply compression if configured
+      if (compressionAlgorithm) {
```

Comment:
We could not repeat publish code by sharing common part. If statement should be there only for payload construction and publish options